src/BenchmarkDotNet.Diagnostics.Windows/Tracing/NativeMemoryLogParser.cs

```diff
@@ -0,0 +1,206 @@
+                    }
+
+                    if (allocs.TryGetValue(data.FreeAddress, out long alloc))
+                    {
+                        nativeLeakSize -= alloc;
+
+                        allocs.Remove(data.FreeAddress);
+                    }
+                };
+                heapParser.HeapTraceReAlloc += delegate(HeapReallocTraceData data)
+                {
+                    if (!start)
+                    {
+                        return;
+                    }
+
+                    // Reallocs that actually move stuff will cause a Alloc and delete event
+                    // so there is nothing to do for those events.  But when the address is
+                    // the same we need to resize
+                    if (data.OldAllocAddress != data.NewAllocAddress)
+                    {
+                        return;
+                    }
+
+                    var allocs = lastHeapAllocs;
+                    if (data.HeapHandle != lastHeapHandle)
+                    {
+                        allocs = CreateHeapCache(data.HeapHandle, heaps, ref lastHeapAllocs, ref lastHeapHandle);
+                    }
+
+                    // This is a clone of the Free code
+                    if (allocs.TryGetValue(data.OldAllocAddress, out long alloc))
+                    {
+                        nativeLeakSize -= alloc;
+
+                        allocs.Remove(data.OldAllocAddress);
+                    }
+
+                    // This is a clone of the Alloc code (sigh don't clone code)
+                    allocs[data.NewAllocAddress] = data.NewAllocSize;
+
+                    nativeLeakSize += data.NewAllocSize;
+                };
+                heapParser.HeapTraceDestroy += delegate(HeapTraceData data)
+                {
+                    if (!start)
+                    {
+                        return;
+                    }
+
+                    // Heap is dieing, kill all objects in it.
+                    var allocs = lastHeapAllocs;
+                    if (data.HeapHandle != lastHeapHandle)
+                    {
+                        allocs = CreateHeapCache(data.HeapHandle, heaps, ref lastHeapAllocs, ref lastHeapHandle);
+                    }
+
+                    foreach (var alloc in allocs.Values)
+                    {
+                        // TODO this is a clone of the free code.
+                        nativeLeakSize -= alloc;
+                    }
+                };
+
+                eventSource.Process();
+
+                logger.WriteLine();
+                logger.WriteLineHeader(LogSeparator);
+                logger.WriteLineInfo($"{benchmarkCase.DisplayInfo}");
+                logger.WriteLineHeader(LogSeparator);
+
+                var totalMemory = totalAllocation / totalOperation;
+                var memoryLeak = nativeLeakSize / totalOperation;
+
+                logger.WriteLine($"Total allocated native memory: {totalMemory.ToSizeStr(SizeUnit.B)}");
```

Comment:
similar as above: it's not total, it's scaled per operation
```suggestion
                logger.WriteLine($"Native memory allocated per single operation: {totalMemory.ToSizeStr(SizeUnit.B)}");
```